### PR TITLE
Problem: missing returning of SignDoc for ledger #130

### PR DESCRIPTION
**Pull request by leejw51crypto**
add toSignDocRaw
this is bytes before hashing, can be used for ledger

### CODE REVIEW

**Review by eddycpc — DISMISSED** [review]
Looks good.

**Review by crypto-matto — DISMISSED** [review]
LGTM~

File: lib/src/transaction/signable.ts [hunk 1]
```diff
@@ -73,42 +73,51 @@ export class SignableTransaction {
...
     }
 
+    /**
+     * Returns the SignDoc of the specified index
+     * @param {number} index index of the signer
+     * @returns {Bytes}
+     * @throws {Error} when index is invalid
+     * @memberof SignableTransaction
+     */
+    public toSignDoc(index: number): Bytes {
```
Comment (tomtau) [inline lib/src/transaction/signable.ts:115]: `toSignDocHash` may be a better name? this doesn't return the signdoc payload, but sha256 of signdoc payload

Comment (leejw51crypto) [reply lib/src/transaction/signable.ts:115]: not bad, i'll change it

**Review by calvinaco — APPROVED** [review]

File: lib/src/transaction/signable.ts (hunk 1 above)
Comment (calvinaco) [reply lib/src/transaction/signable.ts:115]: It will be better to provide this as an alias. coz changing `toSignDoc` may break some codes. @eddycpc

Comment (tomtau) [reply lib/src/transaction/signable.ts:115]: @calvinaco keep the old name with the original functionality (i.e. keep `toSignDoc` as what's in this PR `toSignDocRaw`) and name this new function that's returning the signing digest differently?
as per C4: https://rfc.zeromq.org/spec/42/#26-evolution-of-public-contracts
> Old names SHALL NOT be reused by new contracts.

Comment (leejw51crypto) [reply lib/src/transaction/signable.ts:115]: by npm version, previous published version will not be impacted, though.

**Review by linfeng-crypto — APPROVED** [review]

File: lib/src/transaction/signable.ts (hunk 1 above)
Comment (leejw51crypto) [reply lib/src/transaction/signable.ts]: change the names, and preserved oldname not to break legacy code

Comment (calvinaco) [reply lib/src/transaction/signable.ts]: Yes agree. To make it clear:

`toSignDocument`: New method introduced that returns the normal (Non-hashed) sign doc
`toSignDocumentHash`: New method introduced that returns hashed sign doc
`toSignDoc`: Going to be deprecated method which has the same behaviour as before (same as `toSignDocumentHash`)

Comment (leejw51crypto) [reply lib/src/transaction/signable.ts]: yes, i'm changing unit-test.

File: lib/e2e/transaction.spec.ts [hunk 2]
```diff
@@ -109,8 +109,8 @@ describe('e2e test suite', function () {
             .toSignable();
 
         const signedTx = signableTx
-            .setSignature(0, keyPair.sign(signableTx.toSignDoc(0)))
-            .setSignature(1, keyPair2.sign(signableTx.toSignDoc(1)))
+            .setSignature(0, keyPair.sign(signableTx.toSignDocumentHash(0)))
```
Comment (eddycpc) [inline lib/e2e/transaction.spec.ts:112]: We might have to be careful here, **this is a breaking change for all other API users**: chain desktop wallet and all other folks that have already started integrating.

**Review by eddycpc — CHANGES_REQUESTED** [review]

File: lib/e2e/transaction.spec.ts (hunk 2 above)
Comment (leejw51crypto) [reply lib/e2e/transaction.spec.ts:112]: toSignDoc is still supported, so this pr not break for the time being

Comment (eddycpc) [reply lib/e2e/transaction.spec.ts:112]: Very cool if it's that way, Thanks

**Review by eddycpc — APPROVED** [review]

File: lib/src/transaction/signable.ts (hunk 1 above)
Comment (calvinaco) [reply lib/src/transaction/signable.ts]: cc @eddycpc

**Comment by calvinaco** [comment]
@leejw51crypto is it good to merge?

**Comment by leejw51crypto** [comment]
yes

**Pull request merged**